packages/flutter_tools/lib/src/windows/build_windows.dart

```diff
@@ -19,12 +19,23 @@ import '../globals.dart' as globals;
 import '../migrations/cmake_custom_command_migration.dart';
 import 'visual_studio.dart';
 
+/// These characters appear to be fine: @%()-+_{}[]`~
+const String badCharacters = r"'#!$^&*=|,;<>?";
+
 /// Builds the Windows project using msbuild.
 Future<void> buildWindows(WindowsProject windowsProject, BuildInfo buildInfo, {
   String? target,
   VisualStudio? visualStudioOverride,
   SizeAnalyzer? sizeAnalyzer,
 }) async {
+  final String projectPath = windowsProject.parent.directory.absolute.path;
+  final bool badPath = badCharacters.runes
+      .any((int i) => projectPath.contains(String.fromCharCode(i)));
+  if (badPath) {
+    throwToolExit(
```

Comment:
```
    throwToolExit('Path $projectPath contains invalid characters in "$badCharacters"');
```

Reply:
What is the difference? The newline?

Reply:
yes, this formatting does not follow Flutter's style guide https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#formatting